apps/web/components/booking/AvailableTimes.tsx

```diff
@@ -61,6 +63,10 @@ const AvailableTimes: FC<AvailableTimesProps> = ({
     [isMobile]
   );
 
+  const reserveSlot = (slot: Slot) => {
+    mutation.mutate({ slotUtcDate: slot.time, eventTypeId });
```

Comment:
I think we should reserve a user's timeslot instead of reserving it per event, as timeslots are shared across event types.

For Team Events with Round Robin scheduling, a slot can be booked multiple times and it can prevent booking a slot when in fact it is available.
Similarly, for events with multiple seats, I believe we need to consider the number of seats

Might be taken care of in a follow-up too.

Reply:
Per event type is fine for now. this might open up a can of worms that we don't know yet.

Reply:
do you think per event type is better vs for all links? i think there is a potential for more double bookings if its per event-type only

Reply:
I think @hariombalhara is right, the timeslot must be reserved, it doesn't matter the event.

I propose add the userId to the table and a boolean field to indicate that this revervation is an event's seat, this way we can count the reserve seats and added to the real occupied seats.

What is your opinion about this ?@zomars @PeerRich @alannnc @emrysal

Reply:
Agreed